pnpm/src/switchCliVersion.ts

```diff
@@ -44,31 +43,6 @@ export async function switchCliVersion (config: Config): Promise<void> {
     // We throw this error to prevent an infinite recursive call of the same pnpm version.
     throw new VersionSwitchFail(pm.version, wantedPnpmBinDir)
   }
-
-  // Specify the exact pnpm file path that's expected to execute to spawn.sync()
-  //
-  // It's not safe spawn 'pnpm' (without specifying an absolute path) and expect
-  // it to resolve to the same file path computed above due to the $PATH
-  // environment variable. While that does happen in most cases, there's a
-  // scenario where the wanted pnpm bin dir exists, but no pnpm binary is
-  // present within that directory. If that's the case, a different pnpm bin can
-  // get executed, causing infinite spawn and fork bombing the user. See details
-  // at https://github.com/pnpm/pnpm/pull/8679.
-  const pnpmBinPath = path.join(wantedPnpmBinDir, 'pnpm')
-
-  const { status, error } = spawn.sync(pnpmBinPath, process.argv.slice(2), {
-    stdio: 'inherit',
-    env: {
-      ...process.env,
-      [pnpmEnv.name]: pnpmEnv.value,
-    },
-  })
-
-  if (error) {
-    throw new VersionSwitchFail(pm.version, wantedPnpmBinDir, error)
-  }
-
-  process.exit(status ?? 0)
```

Comment:
If we simply let switchCliVersion code finish without exiting or executing `pnpm` command, later in `main.ts`, pnpm commands will be carried out ([code](https://github.com/pnpm/pnpm/blob/main/pnpm/src/main.ts#L293)). In the local development, this will be `pd` command.  This means the advantages of using this `pnpm` command in main.ts are that
- we can remove duplicated codes
- we can utilize `pd` command in the local development. If we carry out commands in `switchCliVersion`, this won't be possible.